rfcs/0017-tx-valid-since/0017-tx-valid-since.md

```diff
@@ -2,335 +2,192 @@
+* Otherwise, the cell is immature.
+
+The transaction verification fails when any of its input is immature.
+
+![](since-verification.jpg)
+
+### Time Metrics
+
+The metric flag determines how to obtain the cell creation and destruction time, and how to interpret the `value`.
+
+Assume that the `S` is block in which the transaction creating the cell is committed, and `T` is the block where the current transaction is committed. If the transaction is still in the memory pool, `T` is the next to be mined block. `T - i` means the block which block number is  block `T`'s block number minus `i`.
+
+| Metric | Creation Time | Destruction Time | Value |
+| ------ | ------------- | ---------------- | ----- |
+| 00     | block number of S | block number of T | Number of blocks in integer |
+| 01     | epoch number with fraction of S | epoch number with fraction of T | number of epochs in fraction |
+| 10     | timestamp median of \[ S - 37, S - 1 \] | timestamp median of \[ T - 37, T - 1 \] | Number of milliseconds |
+
+#### Block Number
+
+When the metric flag is `00` (Block number):
+
+* The creation time is the block number of `S`.
+* The destruction time is the block number of `T`.
+* The `value` is a unsigned integer. It is a block number when the relative flag is `0` (absolute), and is the number of blocks when of relative flag is `1` (relative).
+
+#### Epoch Number With Fraction
+
+When the metric flag is `01` (Epoch number with fraction):
+
+* The creation time is a rational number `E(S) + I(S) / L(S)`, where
+    * `E(S)` is the epoch number of the epoch that the block `S` is in.
+    * `I(S)` is the block index of `S` in the epoch. The index of the first block in an epoch is 0.
+    * `L(S)` is the epoch length.
+* The destruction time is a rational number `E(T) + I(T) / L(T)`, where
+    * `E(T)` is the epoch number of the epoch that the block `T` is in.
+    * `I(T)` is the block index of `T` in its epoch.
+    * `L(T)` is the epoch length.
+* The `value` is a encoded rational number `E + I / L` the same as the `epoch` field in the block header, where
+
+    ![](../0027-block-structure/epoch.png)
+
+    * `E` is the epoch number part, the three bytes from the lowest bit 0 to 23.
+    * `I` is the block index in the epoch part, the two bytes from the bit 24 to 39.
+    * `L` is the epoch length part, the two bytes from the bit 40 to 55. There's an **exception** here, `L` is at least 1. If `L` is zero, the whole rational number is considered as `E + 0 / 1`.
+
+**Specially**, when the relative flag is `0` (absolute), the `value` is normalized first. But when the relative flag is `1` (relative), the `value` is interpreted as it is and is added to the creation time directly.
+
+**Absolute Epoch With Fraction Value Normalization**: If the relative flag is `0` (absolute)　and `I` is larger than or equal to `L`, the `value` is normalized to `E + 1 + 0 / 1`.
```

Comment:
So, by following this rule, we normalize `100 + 201 / 100` to `101 + 0 / 1`. IMO it is a counter-intuitive behaviour, why can't we just reject these values?

Reply:
This phase is outdated #233

Reply:
Yes, deprecated after #223 is activated. A notice should be added that the rule is removed after #223 is merged. Before that, thia PR should be reviewed based on the current consensus edition.

Only after #223 is activated in testnet and mainnet, and there are no history blocks in testnet and mainnet violating #223, can we remove it from this RFC.